In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import scipy.stats as stats
%matplotlib inline 

In [62]:
df = pd.read_csv('student_drop_out_new_dataset.csv')
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 19 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0                          10000 non-null  int64  
 1   Age                    10000 non-null  float64
 2   Gender                 10000 non-null  object 
 3   Family_Income          10000 non-null  float64
 4   Internet_Access        10000 non-null  object 
 5   Study_Hours_per_Day    10000 non-null  float64
 6   Attendance_Rate        10000 non-null  float64
 7   Assignment_Delay_Days  10000 non-null  int64  
 8   Travel_Time_Minutes    10000 non-null  float64
 9   Part_Time_Job          10000 non-null  object 
 10  Scholarship            10000 non-null  object 
 11  Stress_Index           10000 non-null  float64
 12  GPA                    10000 non-null  float64
 13  Semester_GPA           10000 non-null  float64
 14  CGPA                   10000 non-null  float64
 15  Sem

## Do actual dropout rates amount genders are same or females' dropout rates are higher

1) **Hypothesis testing**:

H0: population_mean_of_female = population_mean_of_male

H1: population_mean_of_female < population_mean_of_male

2) significant level = 0.05
   **Decision rule**
   IF testing point <critical point, reject H0 esle accept H0.


In [42]:
data_male, data_female = df[df['Gender']=='Male'], df[df['Gender'].isin(['Female'])]
df['Gender'].value_counts() 

Gender
Female    5011
Male      4989
Name: count, dtype: int64

Sample size for females is 5011, sample size for males is 4989

In [27]:
nm = len(data_male) 
nf = len(data_female)
print(f" Sample size of male: {nm}")
print(f" Sample size of female: {nf}")

 Sample size of male: 4989
 Sample size of female: 5011


In [35]:
sm = (data_male['Dropout']==1).sum() /nm
sf = (data_male['Dropout']==1).sum()/nf
print(f" Sample male proportion: {pm:.2f}")
print(f" Sample female proportion: {pf:.2f}")

 Sample male proportion: 0.23
 Sample female proportion: 0.23


In [36]:
pp = (nm *pm + nf * pf)/(nm+nf)
qp = 1-pp
print(f"Pooled proportion: {pp:.2f}")

Pooled proportion: 0.23


In [37]:
critical_point = 0-1.645*(pp*qp/nm +pp*qp/nf)**0.5
print(f"Critical point of this study: {critical_point:.2f}")

Critical point of this study: -0.01


In [41]:
testing_point = sf-sm
print(f"Testing point of this study is {testing_point:.3f}")

Testing point of this study is -0.001


**Based on the hypothesis test, there is no statistically significant evidence to suggest a difference in dropout rates between males and females. Therefore, we fail to reject the null hypothesis.**

## DO ACTUAL DROPOUT RATES ARE SAME AMOUNG SCHOLARSHIP TYPES, OR THE DROPOUT RATE OF STUDENTS WHO HAVE SCHOLARSHIP IS LOWER.

**Hypothesis testing**

H0: population_mean_scholarship_yes = population_mean_scholarship_no

H1: population_mean_scholarship_yes < population_mean_scholarship_no

**Decision Rule** at significant level is 0.05

If testing_point<critical_point, reject H0 else accept H0

In [59]:
dy = df[df['Scholarship']=='Yes']
dn = df[df['Scholarship']=='No']

In [47]:
ny = len(dy)
nn = len(dn)

In [53]:
sy = (dy['Dropout']==1).sum()/ny
sn = (dn['Dropout']==1).sum()/nn
print(f"Sample proportion where Scholarship is Yes: {sy:.3f}")
print(f"Sample proportion where Scholarship is No : {sn:.3f}")

Sample proportion where Scholarship is Yes: 0.232
Sample proportion where Scholarship is No : 0.237


In [54]:
pp_s = (ny*sy+nn*sn)/(ny+nn)
qp_s=1-pp_s
print(f"Pooled proportion: {pp:.3f}")

Pooled proportion: 0.235


In [58]:
testing_point = sy-sn
critical_point =0-1.645* (pp_s*qp_s/ny +pp_s*qp_s/nn)**0.5
print(f"Testing point of this study : {testing_point:.3f}")
print(f"Critical point of this study: {critical_point:.3f}")

Testing point of this study : -0.006
Critical point of this study: -0.015


In [57]:
if testing_point < critical_point:
    print("Reject H0")
else:
    print('Accept H0')

Accept H0


**Based on the hypothesis test, there is no statistically significant evidence to suggest a difference in dropout rates between scholarship types. Therefore, we fail to reject the null hypothesis.**